src/rf/apps/home/views.py

```diff
@@ -205,9 +205,24 @@ def _get_layer_models(request, crit=None):
         qs = qs.filter(crit)
 
     filtered_layers = LayerFilter(request.GET, queryset=qs)
-    paginator = Paginator(filtered_layers, RESULTS_PER_PAGE)
 
     page = request.GET.get('page')
+    page_size = request.GET.get('page_size')
+
+    results_per_page = RESULTS_PER_PAGE
+    if page_size:
+        try:
+            page_size = int(page_size)
+            num_layers = filtered_layers.count()
+            if page_size == 0:
+                if num_layers > 0:
+                    results_per_page = num_layers
```

Comment:
The purpose of pagination is to limit the amount of data transferred to the client at one time to reduce the amount of load on the DB. Allowing anybody to subvert that limitation by setting page_size to 0 is questionable. 

In any case, we could reduce the number of queries we need to perfom by hard-coding a reasonable upper limit when page_size is 0. It's difficult to imagine anybody having more than 10 or 20 simultaneous layers uploading at a given time. An upper limit of 50, 100, or 1000 should suffice.